frontend/admin/src/js/components/user/GeneralInformationTab/AddToPoolDialog.tsx

```diff
@@ -0,0 +1,231 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+
+import Dialog from "@common/components/Dialog";
+import Button from "@common/components/Button";
+import { toast } from "@common/components/Toast";
+import { getFullNameHtml } from "@common/helpers/nameUtils";
+import { getFullPoolAdvertisementTitle } from "@common/helpers/poolUtils";
+import { Input, Select } from "@common/components/form";
+import { errorMessages } from "@common/messages";
+import { currentDate } from "@common/helpers/formUtils";
+import {
+  AdvertisementStatus,
+  CreatePoolCandidateAsAdminInput,
+  Pool,
+  PoolCandidate,
+  useCreatePoolCandidateMutation,
+  User,
+} from "../../../api/generated";
+
+type FormValues = {
+  poolId: Pool["id"];
+  expiryDate: PoolCandidate["expiryDate"];
+};
+
+export interface AddToPoolDialogProps {
+  user: User;
+  pools: Pool[];
+}
+
+export const AddToPoolDialog: React.FC<AddToPoolDialogProps> = ({
+  user,
+  pools,
+}) => {
+  const intl = useIntl();
+  const [open, setOpen] = React.useState(false);
+  const methods = useForm<FormValues>();
+
+  const [, executeMutation] = useCreatePoolCandidateMutation();
```

Comment:
Could we pop off `fetching` here and use it to disable the button and update the text content so the user knows something is happening?

> **Note**
> Applies to all three forms

```tsx
const [{ fetching }, executeMutation] = useCreatePoolCandidateMutation();

...

<Button disabled={fetching}>
  {fetching ? intl.formatMessage({
    defaultMessage: "Saving...",
  }) : intl.formatMessage({
     defaultMessage: "Add to new pool",
  })}
</Button>
```